packages/dds/matrix/src/test/memory/sharedMatrix.spec.ts

```diff
@@ -23,221 +23,59 @@ import { createLocalMatrix, type TestMatrixOptions } from "../utils.js";
  */
 
 // TODOs (AB#46340):
-// - single helper function with before and after hooks for setup and teardown
 // - unify with time measurement tests (in terms of API)
 
-/**
- * Initializes a SharedMatrix for testing.
- * @remarks Includes initialization of the undo/redo stack, as well as mock event subscriptions.
- */
-function createMatrix(options: TestMatrixOptions): {
-	/**
-	 * The initialized matrix.
-	 */
-	matrix: ISharedMatrix;
-
-	/**
-	 * The undo/redo stack manager for the matrix.
-	 */
-	undoRedoStack: UndoRedoStackManager;
-
-	/**
-	 * Cleanup function to run after the test to close the matrix and release resources.
-	 */
-	cleanUp: () => void;
-} {
-	const matrix = createLocalMatrix(options);
-
-	// Configure event listeners
-	const eventListeners: IMatrixConsumer<string> = {
-		rowsChanged: () => {},
-		colsChanged: () => {},
-		cellsChanged: () => {},
-	};
-	matrix.openMatrix(eventListeners);
-
-	// Configure undo/redo
-	const undoRedoStack = new UndoRedoStackManager();
-	matrix.openUndo(undoRedoStack);
-
-	const cleanUp = (): void => {
-		matrix.closeMatrix(eventListeners);
-	};
-
-	return {
-		matrix,
-		undoRedoStack,
-		cleanUp,
-	};
-}
-
-/**
- * {@link createBenchmark} options.
- */
-interface BenchmarkOptions extends TestMatrixOptions {
-	/**
-	 * The title of the benchmark test.
-	 */
-	readonly title: string;
-
-	/**
-	 * The operation to be measured.
-	 */
-	readonly operation: (matrix: ISharedMatrix) => void;
-}
-
 /**
  * Creates a benchmark for operations on a SharedMatrix.
  */
-function createBenchmark({
```

Comment:
Not for reviewers: all usages of this immediate wrapped it in a call to `benchmarkMemory`. Updated to just do that here instead (which also makes the test setup here more like the execution time tests).